FRAuth/FRAuth/DeviceBinding/KeyAware.swift

```diff
@@ -0,0 +1,99 @@
+// 
+//  KeyAware.swift
+//  FRAuth
+//
+//  Copyright (c) 2022 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+
+import Foundation
+import CommonCrypto
+import FRCore
+
+///Helper struct to generate and sign the keys
+struct KeyAware {
+    
+    var userId: String
+    var timeout = 60
+    private var keySize = 2048
+    private var keyType = kSecAttrKeyTypeRSA
+    private var key: String
+    
+    
+    /// Initizialier for KeyAware
+    /// - Parameter userId: user id for which key pair is to be generated
+    init(userId: String) {
+        self.userId = userId
+        self.key = KeyAware.getKeyAlias(keyName: userId)
+    }
+    
+    
+    /// Query attributes dictionary for generating key pair
+    func keyBuilderQuery() -> [String: Any] {
+        var query = [String: Any]()
+        
+        query[String(kSecAttrKeyType)] = String(keyType)
+        query[String(kSecAttrKeySizeInBits)] = keySize
+        
```

Comment:
i belive the secure enclave the part will be somewhere here 

` #if !targetEnvironment(simulator)
        // If the device supports Secure Enclave, create a keypair using Secure Enclave TokenID
        if SecuredKey.isAvailable() {
            query[String(kSecAttrTokenID)] = String(kSecAttrTokenIDSecureEnclave)
            let accessControl = SecAccessControlCreateWithFlags(kCFAllocatorDefault, accessibility.rawValue as CFString, [.privateKeyUsage, .userPresence] , nil)!
            keyAttr[String(kSecAttrAccessControl)] = accessControl
        }
        #endif`